### PR TITLE
Add more draw functions to scripted gauge

### PR DESCRIPTION
**Pull request by wookieejedi**
This adds the following functions to the scripted gauge API: `drawCircle`, `drawRect`, and `drawLine`. Also note that I created an additional function, for `HUDGauge:renderCircle_unfilled`, because the precedent shown by having two functions for `gr_unfilled_circle` and `gr_circle` and adding a fourth argument to `renderCircle` seemed to affect a lot of other functions.

### CODE REVIEW

File: code/scripting/api/objs/hudgauge.cpp [hunk 1]
```diff
@@ -83,14 +84,115 @@ ADE_FUNC(drawString,
...
+		gauge->renderLine(fl2i(gauge_x + x1), fl2i(gauge_y + y1), fl2i(gauge_x + x2), fl2i(gauge_y + y1));
+		gauge->renderLine(fl2i(gauge_x + x1), fl2i(gauge_y + y2), fl2i(gauge_x + x2), fl2i(gauge_y + y2));
+		gauge->renderLine(fl2i(gauge_x + x1), fl2i(gauge_y + y1), fl2i(gauge_x + x1), fl2i(gauge_y + y2));
+		gauge->renderLine(fl2i(gauge_x + x2), fl2i(gauge_y + y1), fl2i(gauge_x + x2), fl2i(gauge_y + y2));
+	}
+
+	return ADE_RETURN_TRUE;
+}
+
+ADE_FUNC(drawImage, l_HudGaugeDrawFuncs, "texture handle Texture, [number X=0, Y=0]",
```
Comment (asarium) [inline code/scripting/api/objs/hudgauge.cpp]: I think you'll need to rebase your branch to get it up to date with the newest changes from your other pull request.

File: code/hud/hud.cpp [hunk 2]
```diff
@@ -947,6 +947,28 @@ void HudGauge::renderCircle(int x, int y, int diameter)
 	gr_reset_screen_scale();
 }
 
+void HudGauge::renderCircle_unfilled(int x, int y, int diameter)
```
[the remark this answers is not in the document]
Comment (wookieejedi) [reply code/hud/hud.cpp]: Yeah, I tried do this:
```void HudGauge::renderCircle(int x, int y, int diameter, bool filled=true) 
{
	int nx = 0, ny = 0;

	if ( gr_screen.rendering_to_texture != -1 ) {
		gr_set_screen_scale(canvas_w, canvas_h, -1, -1, target_w, target_h, target_w, target_h, true);
	} else {
		if ( reticle_follow ) {
			nx = HUD_nose_x;
			ny = HUD_nose_y;

			gr_resize_screen_pos(&nx, &ny);
			gr_set_screen_scale(base_w, base_h);
			gr_unsize_screen_pos(&nx, &ny);
		} else {
			gr_set_screen_scale(base_w, base_h);
		}
	}
	if (filled) {
		gr_circle(x + nx, y + ny, diameter);
	} else {
		gr_unfilled_circle(x + nx, y + ny, diameter);
	}
	
	gr_reset_screen_scale();
}```
and also changing `void renderCircle(int x, int y, int diameter, bool filled=true);` but I'm still getting errors like `hud\hud.cpp(927): error C2572: 'HudGauge::renderCircle': redefinition of default argument: parameter 1` and `hud\hud.h(321): note: see declaration of 'HudGauge::renderCircle'`. So it seems like I need to change something else?

File: code/scripting/api/objs/hudgauge.cpp (hunk 1 above)
Comment (wookieejedi) [reply code/scripting/api/objs/hudgauge.cpp]: Noted, thanks!

File: code/hud/hud.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (wookieejedi) [reply code/hud/hud.cpp]: Or with `void renderCircle(int x, int y, int diameter, bool filled);` I get the error message `radar\radar.cpp(185): error C2660: 'HudGauge::renderCircle': function does not take 3 arguments`

[the remark this answers is not in the document]
Comment (wookieejedi) [reply code/hud/hud.cpp]: Will do!

**Pull request merged**